src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/FontCache/FontSourceCollection.cs

```diff
@@ -154,16 +154,16 @@ private void SetFontSources()
                                             if (Path.GetFileName(fileName) == fileName)
                                                 fileName = Path.Combine(Util.WindowsFontsLocalPath, fileName);
 
-                                            fontPaths[fileName] = null;
+                                            fontPaths.Add(fileName);
                                         }
                                     }
                                 }
 
                                 foreach (string file in Directory.GetFiles(_uri.LocalPath))
                                 {
-                                    fontPaths[file] = null;
+                                    fontPaths.Add(file);
                                 }
```

Comment:
Why not do directly `fontPaths.UnionWith(Directory.GetFiles(_uri.LocalPath))`?

Reply:
@ThomasGoulet73 any comments ?

Reply:
@halgab and maybe even better use `EnumerateFiles` ?

Reply:
I forgot about this. I did a quick benchmark with both of your suggestions and it looks like using Directory.EnumerateFiles shows a nice improvement and that UnionWith is the same as a foreach when using Directory.EnumerateFiles:
|                  Method |     Mean |   Error |  StdDev | Ratio | RatioSD |   Gen0 | Allocated | Alloc Ratio |
|------------------------ |---------:|--------:|--------:|------:|--------:|-------:|----------:|------------:|
|         ForEachGetFiles | 174.7 μs | 3.42 μs | 6.75 μs |  1.00 |    0.00 | 3.1738 |  10.09 KB |        1.00 |
|   ForEachEnumerateFiles | 161.9 μs | 2.97 μs | 2.77 μs |  0.90 |    0.05 | 2.6855 |   8.65 KB |        0.86 |
|       UnionWithGetFiles | 166.6 μs | 3.29 μs | 5.02 μs |  0.95 |    0.05 | 3.1738 |  10.13 KB |        1.00 |
| UnionWithEnumerateFiles | 161.3 μs | 2.96 μs | 3.29 μs |  0.90 |    0.05 | 2.6855 |   8.65 KB |        0.86 |

I decided to go with using UnionWith and Directory.EnumerateFiles since it's cleaner and faster (Directory.GetFiles just wraps the result of Directory.EnumerateFiles in an array and since we just add the result to an hashset, the array is just wasted memory and makes it iterate the result twice).